LifeSpace/Debug/LogManager.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  LogStore.swift
+//  LifeSpace
+//
+//  Created by Vishnu Ravi on 11/8/24.
+//
+
+import Foundation
+import OSLog
+import Spezi
+import SwiftUI
+
+actor LogManager: Module, DefaultInitializable, EnvironmentAccessible {
+    @Application(\.logger) private var logger
+    
+    func query(
+        startDate: Date? = nil,
+        endDate: Date? = nil,
+        logType: OSLogEntryLog.Level? = nil
+    ) -> String {
+        do {
+            let store = try OSLogStore(scope: .currentProcessIdentifier)
+            let position: OSLogPosition
+            if let startDate = startDate {
+                position = store.position(date: startDate)
+            } else {
+                position = store.position(timeIntervalSinceLatestBoot: 1)
+            }
+            
+            let logs = try store.getEntries(at: position).compactMap { $0 as? OSLogEntryLog }
+            
+            return logs
+                .filter { logEntry in
+                    /// Filter by subsystem
+                    guard logEntry.subsystem == Bundle.main.bundleIdentifier else {
+                        return false
+                    }
+                    
+                    /// Filter by log type if specified
+                    if let logType, logEntry.level != logType {
+                        return false
+                    }
+                    
+                    /// Filter by date range if specified
+                    if let startDate, logEntry.date < startDate {
+                        return false
+                    }
+                    
+                    if let endDate, logEntry.date > endDate {
+                        return false
+                    }
+                    
+                    return true
+                }
+                .map { "[\($0.date.formatted())] [\($0.category)] \($0.composedMessage)" }
```

Comment:
Would be really cool to build attributed strings for the visualization to e.g., highlight categories in red for errors, yellow for warnings and others in the future. Or it might even be good to pass out a custom struct here and do the formatting in the UI components to ensure that others could reuse this in different ways.

Reply:
I've updated this to just return the `OSLogEntryLog`s as an array and the formatting is done in the UI components.